library/src/lib/calendar/calendar.component.ts

```diff
@@ -65,120 +68,170 @@ let calendarUniqueId: number = 0;
     @Input()
     startingDayOfWeek: WeekDaysNumberRange = 0;
 
+    /** Fired when the date input value is invalid. */
     @Output()
     isInvalidDateInput: EventEmitter<any> = new EventEmitter();
 
+    /** @hidden Used when this calendar is for a date time picker component. For internal use. */
     @Input()
     isDateTimePicker: boolean = false;
 
+    /** @hidden Whether the date is invalid. Internal use */
     invalidDate: boolean = false;
 
+    /** @hidden Whether to show the month selection grid on the calendar. Internal use */
     showCalendarMonths: boolean = false;
+    /** @hidden Whether to show the year selection grid on the calendar. Internal use */
     showCalendarYears: boolean = false;
+    /** @hidden Whether to show the date selection grid on the calendar. Internal use */
     showCalendarDates: boolean = true;
 
+    /** @hidden For i18n, the list of month short names. Internal use */
     monthsShortName: string[];
+    /** @hidden For i18n, the list of month full names. Internal use */
     monthsFullName: string[];
 
+    /** @hidden For i18n, the list of weekday names. Internal use */
     weekDays: string[];
+
+    /** @hidden The typical number of days in each month. Internal use */
     daysPerMonth = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
 
+    /** @hidden The displayed grid of calendar days. Internal use */
     calendarGrid: CalendarDay[][] = [];
+    /** @hidden The years to display in the year selection grid. Internal use */
     calendarYearsList: number[] = [];
 
+    /** @hidden Today's date. Internal use */
     today: Date = new Date();
+    /** @hidden Today's month. Internal use */
     todayMonth = this.today.getMonth();
+    /** @hidden Today's year. Internal use */
     todayYear = this.today.getFullYear();
 
+    /** @hidden Date used to navigate the calendar. Not to be confused with selectedDay which is the ngModel. Internal use */
     date: Date = new Date();
+    /** @hidden Month used to navigate the calendar.Internal use */
     month: number = this.date.getMonth();
+    /** @hidden Name of the month currently displayed. Internal use */
     monthName: string;
+    /** @hidden Year used to navigate the calendar. Internal use */
     year: number = this.date.getFullYear();
+    /** @hidden Day of month used to navigate the calendar. Internal use */
     day = this.date.getDate();
 
+    /** @hidden Number (0-11) of the selected month used to navigate the calendar. Internal use */
     selectedMonth: number;
+
+    /** @hidden The first year to be displayed in the list of selectable years. Internal use */
     firstYearCalendarList: number = this.year;
+
+    /** @hidden The first year to be displayed in the list of selectable years. Internal use */
     selectCounter: number = 0;
 
+    /** The currently selected CalendarDay model */
     @Input()
     selectedDay: CalendarDay = {
         date: null
     };
+    /** Fired when the selectedDay model changes */
```

Comment:
this is very vague and could be applied to most events - the description should also involve the context in which this model changes, which in this case is when a new day is selected